Geometry/HcalAlgo/plugins/DDHCalAngular.cc

```diff
@@ -32,37 +38,44 @@ void DDHCalAngular::initialize(const DDNumericArguments & nArgs,
   n           = int (nArgs["n"]);
   startCopyNo = int (nArgs["startCopyNo"]);
   incrCopyNo  = int (nArgs["incrCopyNo"]);
-  LogDebug("HCalGeom") << "DDHCalAngular debug: Parameters for positioning-- "
-		       << n << " copies in " << rangeAngle/CLHEP::deg 
-		       << " from " << startAngle/CLHEP::deg << "\tShifts " 
-		       << shiftX << ", " << shiftY 
-		       << " along x, y axes; \tZoffest " << zoffset
-		       << "\tStart and inremental copy nos " << startCopyNo 
-		       << ", " << incrCopyNo;
-
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalAngular: Parameters for positioning "
+			       << "-- " << n << " copies in " 
+			       << convertRadToDeg(rangeAngle) << " from " 
+			       << convertRadToDeg(startAngle) << "\tShifts " 
+			       << shiftX << ", " << shiftY 
+			       << " along x, y axes; \tZoffest " << zoffset
+			       << "\tStart and inremental copy nos " 
+			       << startCopyNo << ", " << incrCopyNo;
+#endif
   rotns       = sArgs["RotNameSpace"];
   idNameSpace = DDCurrentNamespace::ns();
   childName   = sArgs["ChildName"]; 
+#ifdef EDM_ML_DEBUG
   DDName parentName = parent().name(); 
-  LogDebug("HCalGeom") << "DDHCalAngular debug: Parent " << parentName 
-		       << "\tChild " << childName << "\tNameSpace "
-		       << idNameSpace << "\tRotation Namespace " << rotns;
+  edm::LogVerbatim("HCalGeom") << "DDHCalAngular: Parent " << parentName 
+			       << "\tChild " << childName << "\tNameSpace "
+			       << idNameSpace << "\tRotation Namespace "
+			       << rotns;
+#endif
 }
 
 void DDHCalAngular::execute(DDCompactView& cpv) {
 
   double dphi   = rangeAngle/n;
-  double phi    = startAngle;
+  double phix   = startAngle;
   int    copyNo = startCopyNo;
+  double theta  = 90._deg;
 
   for (int ii=0; ii<n; ii++) {
-
-    double phideg = phi/CLHEP::deg;
-    int    iphi;
-    if (phideg > 0) iphi = int(phideg+0.1);
-    else            iphi = int(phideg-0.1);
-    if (iphi >= 360) iphi   -= 360;
-    phideg = iphi;
+    double phideg = convertRadToDeg(phix);
```

Comment:
hi @bsunanda - there are several logical changes in this stanza of code. are they intentional?

Reply:
Yes they are. I checked the outcome by running in debug mode and this does not change the results. The HO geometry which is affected by this code gives identical results - I did not explicitly look at the rotation matrix but the input to the matrix - they are the same except may be some rounding errors due to difference in CLHEP::deg and convertRadToDeg or ._deg methods

Reply:
I don't think any tiny difference between `CLHEP::deg` and `_deg` can cause any noticeable changes, and certainly not the thousands of differences seen in the comparison tests. I don't understand it.

Reply:
This PR does contain substantial changes to rounding, but it would be troubling if they could cause the changes seen in the comparison tests.